provider/src/main/kotlin/org/gradle/kotlin/dsl/accessors/CodeGenerator.kt

```diff
@@ -22,142 +22,142 @@ import org.jetbrains.kotlin.lexer.KtTokens
 
 internal
 fun ProjectSchema<TypeAccessibility>.forEachAccessor(action: (String) -> Unit) {
-    extensions.forEach { (name, type) ->
-        extensionAccessorFor(name, type)?.let(action)
+    extensions.map(::typedAccessorSpecFor).forEach { spec ->
+        extensionAccessorFor(spec)?.let(action)
     }
-    conventions.forEach { (name, type) ->
-        if (name !in extensions) {
-            conventionAccessorFor(name, type)?.let(action)
+    conventions.map(::typedAccessorSpecFor).forEach { spec ->
+        if (spec.name.original !in extensions) {
```

Comment:
:x: `conventions.filterKeys { it !in extensions }.forEach { ...` instead of the complex if condition.